client/state/comments/reducer.js

```diff
@@ -24,169 +24,165 @@ import {
-											.concat( Immutable.fromJS( action.comments ) );
-
-				if ( ! action.skipSort ) {
-					newComments = newComments.sort( ( a, b ) => new Date( b.get( 'date' ) ) - new Date( a.get( 'date' ) ) );
-				}
-
-				return newComments;
-			} );
-
+			return {
+				...state,
+				[ `${ siteId }-${ postId }` ]: ! skipSort ? sortBy( action.comments, [ 'date' ] ) : action.comments
+			};
 		case COMMENTS_REMOVE:
-			return updateSpecificState( state, action, ( comments = Immutable.List() ) =>
-				comments.filter( ( comment ) => comment.get( 'ID' ) !== action.commentId )
-			);
+			return {
+				...state,
+				[ `${ siteId }-${ postId }` ]: filter( state[ `${ siteId }-${ postId }` ], { ID: commentId } )
+			};
 		case COMMENTS_LIKE:
-			return updateSpecificState( state, action, ( comments = Immutable.List() ) =>
-				updateExistingIn( comments,
-					( comment ) => comment.get( 'ID' ) === action.commentId,
-					getLikeSetter( { like: true } )
-				)
-			);
+			return {
+				...state,
+				[ `${ siteId }-${ postId }` ]: map( state[ `${ siteId }-${ postId }` ], comment => {
+					if ( comment.ID !== commentId ) {
+						return comment;
+					}
+
+					return {
+						...comment,
+						i_like: true,
+						like_count: comment.like_count++
+					};
+				} )
+			};
 		case COMMENTS_LIKE_UPDATE:
-			return updateSpecificState( state, action, ( comments = Immutable.List() ) =>
-				updateExistingIn( comments,
-					( comment ) => comment.get( 'ID' ) === action.commentId,
-					getLikeSetter( { like: action.iLike, likeCount: action.likeCount } )
-				)
-			);
+			const { iLike, likeCount } = action;
+			return {
+				...state,
+				[ `${ siteId }-${ postId }` ]: map( state[ `${ siteId }-${ postId }` ], comment => {
+					if ( comment.ID !== commentId ) {
+						return comment;
+					}
+
+					return {
+						...comment,
+						i_like: iLike,
+						like_count: likeCount
+					};
+				} )
+			};
 		case COMMENTS_UNLIKE:
-			return updateSpecificState( state, action, ( comments = Immutable.List() ) =>
-				updateExistingIn( comments,
-					( comment ) => comment.get( 'ID' ) === action.commentId,
-					getLikeSetter( { like: false } )
-				)
-			);
+			const { commentId } = action;
+			return {
+				...state,
+				[ `${ siteId }-${ postId }` ]: map( state[ `${ siteId }-${ postId }` ], comment => {
+					if ( comment.ID !== commentId ) {
+						return comment;
+					}
+
+					return {
+						...comment,
+						i_like: false,
+						like_count: comment.like_count--
+					};
+				} )
+			};
 		case COMMENTS_ERROR:
-			return updateSpecificState( state, action, ( comments = Immutable.List() ) =>
-				updateExistingIn( comments,
-					( comment ) => comment.get( 'ID' ) === action.commentId,
-					( comment ) => comment.set( 'placeholderState', PLACEHOLDER_STATE.ERROR )
-										.set( 'placeholderError', action.error )
-				)
-			);
-		case SERIALIZE:
-			return {};
-		case DESERIALIZE:
-			return Immutable.Map();
```

Comment:
We still need to handle the SERIALIZE/DESERIALIZE cases here, since persistence is currently opt-out. I'm working on changing this, but it probably won't land in time to be useful for us. 

`createReducer` usually handles this, but with a bare reducer we'll need to return initial state to opt-out.
```
case SERIALIZE:
   return {};
case DESERIALIZE:
   return {};
```